src/Symfony/Component/Serializer/Normalizer/BackedEnumNormalizer.php

```diff
@@ -0,0 +1,85 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Serializer\Normalizer;
+
+use Symfony\Component\Serializer\Exception\InvalidArgumentException;
+use Symfony\Component\Serializer\Exception\NotNormalizableValueException;
+
+/**
+ * Normalizes a {@see \BackedEnum} enumeration to a string or an integer.
+ *
+ * @author Alexandre Daubois <alex.daubois@gmail.com>
+ */
+final class BackedEnumNormalizer implements NormalizerInterface, DenormalizerInterface, CacheableSupportsMethodInterface
+{
+    /**
+     * {@inheritdoc}
+     *
+     * @throws InvalidArgumentException
+     *
+     * @return int|string
+     */
+    public function normalize($object, string $format = null, array $context = [])
+    {
+        if (!$object instanceof \BackedEnum) {
+            throw new InvalidArgumentException('The data must belong to a backed enumeration.');
+        }
+
+        return $object->value;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function supportsNormalization($data, string $format = null)
+    {
+        return $data instanceof \BackedEnum;
+    }
+
+    /**
+     * {@inheritdoc}
+     *
+     * @throws NotNormalizableValueException
+     */
+    public function denormalize($data, string $type, string $format = null, array $context = [])
+    {
+        if (!is_subclass_of($type, '\BackedEnum')) {
+            throw new InvalidArgumentException('The data must belong to a backed enumeration.');
+        }
+
+        if ('' === $data || null === $data) {
+            throw new NotNormalizableValueException('The data is either an empty string or null, you should pass a string that can be parsed as an enumeration case of type '.$type.'.');
```

Comment:
the message is wrong. You must pass an int or a string, not always a string (depending on the type backing the enum)